### PR TITLE
added form for creating tests

### PR DESCRIPTION
**Pull request by katarinarosiak**
Co-authored-by: Scott Graham <scttgrhm7+public@gmail.com>
Co-authored-by: Katarina Rosiak <katarinarosiak@gmail.com>
Co-authored-by: Miles Abbason <miles.abbason@gmail.com>
Co-authored-by: Tim Dronkers <tdronkers@gmail.com>

There are three views at the moment: 
- Home: this is where the result of the latest test will be displayed later on (not done yet)
- Create Test page: contains a form to create and schedule tests. There is no validation and there is a possibility to add only one assertion at the moment. There are also limited options for locations, comparison, and frequency of schedule. (Is that something we want to query from DB?). 
Pressing the 'Create' button will trigger EventBridge to create a rule that will invoke a Lambda on a certain schedule. (Remove the rule after testing!)
-   Test Results page: The page will contain a list of JSON objects representing test results. This is automatically filled on load. Right now the table is empty so there are no results. 

To do next:
- request data for a single (the latest) test result and display on the home page. 

Screenshot from 2022-07-12 23-00-25
Screenshot from 2022-07-12 22-08-37
Screenshot from 2022-07-12 22-08-24

### CODE REVIEW

File: src/components/CreateTest.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    const testData = {
+      "test": {
+        "title": title,
+        "locations": [locationValue],
+        "minutesBetweenRuns": Number(frequency),
+        "type": "API",
+        "httpRequest": {
```
Comment (tdronkers04) [inline src/components/CreateTest.js:54]: /api/tests route is expecting `method` property instead of `http_method`

File: src/App.js [hunk 2]
```diff
@@ -1,61 +1,20 @@
...
-			<button onClick={handleScheduleTest}>Schedule 1 min</button>
-			<button onClick={handleCreateTest}>Create Test</button>
-		</div>
-	)
+  return (
+    <div>
+      <Routes>
+        <Route path="/" element={<Home />}></Route>
+        <Route index element={<Home />}></Route>
+        <Route path="/TestsResults" element={<TestsResults />}></Route>
```
Comment (scogra17) [inline src/App.js:13]: Wondering if there's any guidance on "proper" casing for url paths? Could be perfectly acceptable as is, but I'm just not used to seeing PascalCase in a path

File: src/components/CreateTest.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+          "http_method": method,
```
Comment (scogra17) [inline src/components/CreateTest.js:55]: I think TestRunner is now expecting just `method`

File: src/services/ApiClient.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  if (response && response.data && response.data.error) {
+    console.error(`HTTP Error: ${response.data.error}`);
+  } else {
+    console.error("Error: ", errorResponse);
+  }
+};
+
+axios.defaults.headers.common["X-Requested-With"] = "XMLHttpRequest";
+axios.defaults.headers.common["Accept"] = "application/json";
```
Comment (scogra17) [inline src/services/ApiClient.js:17]: Did these fix a bug? Just curious

**Comment by scogra17** [comment]
Looks great overall!! Nice work @katarinarosiak. I've added a few comments, hoping they aren't too nit picky :| 

I approved-- feel free to ignore my suggestions if you don't agree!

File: src/components/CreateTest.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+import RadioButtons from './form/RadioButtons'
+import apiClient from '../services/ApiClient';
+import { useState } from 'react';
+
+const CreateTest = () => {
+
+	//locations, source, comparison are hardcoded for now, ideally 
+	// we should get them from DB 
+	const locations = [
+    { label: 'us-east-2', value: 'us-east-2' },
```
Comment (mabbason) [inline src/components/CreateTest.js:13]: Would it make sense to collapse these two separate properties into one? I see that they are used in the Dropdown component but maybe still could use one property?

File: src/services/ApiClient.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import axios from "axios";
+
+const URL = "http://testscrud-env.eba-fpb5kcf8.us-east-1.elasticbeanstalk.com";
+
+function logError(errorResponse) {
+  const response = errorResponse.response;
+
+  if (response && response.data && response.data.error) {
```
Comment (mabbason) [inline src/services/ApiClient.js:9]: I wonder if this is where the 'optional' chaining operator could be used? I have always wanted to use it... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining idk tho?

**Review by tdronkers04 — APPROVED** [review]

File: src/App.js (hunk 2 above)
Comment (katarinarosiak) [reply src/App.js:13]: good point. Thanks

File: src/components/CreateTest.js (hunk 3 above)
Comment (katarinarosiak) [reply src/components/CreateTest.js:55]: perfect. Will change that.

File: src/components/CreateTest.js (hunk 5 above)
Comment (katarinarosiak) [reply src/components/CreateTest.js:13]: totally! Done :)

File: src/services/ApiClient.js (hunk 6 above)
Comment (katarinarosiak) [reply src/services/ApiClient.js:9]: never seen this one before! I can play around and see if it works

**Pull request merged**